framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -351,6 +355,7 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 		await deleteBlocksAfterHeight(
 			this.processorModule,
 			this.blocks,
+			this.logger,
```

Comment:
would you like to keep the logger here? I thought it was for debugging reasons?

Reply:
We needed logs inside this function to trace which blocks are being deleted. I will remove the need of passing dependencies as function aarguments as soon as I move the logic to Base Synchronizer (https://github.com/LiskHQ/lisk-sdk/pull/4537#discussion_r348986838)